### PR TITLE
v3 - Chakra-UI v1 & Next.js v10

### LINKED ISSUE #187: Chakra-UI v1 🤝 Next.js v10
Meta-issue to list the things to do to integrate the new features & evolutions of both Chakra-UI & Next.js

## Chakra-UI v1
- [x] Simplify `_app.tsx`: use the new `ChakraProvider` with integrated CSS reset & global styles
- [x] Check out the [list of codemods to apply](https://next.chakra-ui.com/docs/migration)

## Next.js v10
- [x] Integrate the new `Image` component
- [x] Mark support for `as` in Links as optional

### PR DESCRIPTION
**Pull request by franky47**
- [x] Port existing components to Chakra-UI v1 API
- [x] Simplify _app.ts
- [x] ~Introduce _document.ts to add colorMode persistance~
- [x] ~Add shorthands for `Card` layout types: `<Card.Flex>`, `<Card.Stack>` etc..~ -> No need, replaced by the `as` prop
- [x] Add Image component (see #187)
- [ ] Use [`forwardRef`](https://chakra-ui.com/guides/as-prop#option-1-using-forwardref-from-chakra-uireact) for custom components
- [ ] Mark support for `as` in Links as optional
- [ ] Handle route links in menus (#221)

Documentation
- [ ] Migration guide
- [ ] OutgoingLink `rel` options

Closes #187, #221.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -72,6 +72,9 @@
     "verbose": true,
     "preset": "ts-jest/presets/js-with-ts",
     "testEnvironment": "jsdom",
+    "setupFiles": [
+      "./test/env.js"
```
Comment (franky47) [inline package.json:80]: Note: This is a workaround for Next.js image config not having a default, should be resolved in the next version:
https://github.com/vercel/next.js/discussions/18373#discussioncomment-114212

**Comment by trevorblades** [comment]
Stoked for this release! Is there any way you could publish a prerelease version off of this branch? It seems like the repo is set up to auto deploy, but I thought I'd ask anyway. If the answer is no, I'm happy to be patient 😇

**Comment by franky47** [comment]
Hey Trevor, that sounds like a good idea, I'll look into how to do this. If you have experience with semantic-release, any help is welcome!

**Comment by franky47** [comment]
@trevorblades you can give it a try: `@47ng/chakra-next@3.0.0-beta.1`, feel free to open an issue if you encounter any problems. Breaking changes are listed in the [changelog](https://github.com/47ng/chakra-next/releases/tag/v3.0.0-beta.1).

_**Note to self:** deploying pre-releases is extremely simple with `semantic-release`: enable CD to run on the `beta` branch, then point that branch to whatever you want to deploy, it will semver up and handle dist-tags automatically. 🚀_

**Comment by trevorblades** [comment]
That's great to hear @franky47! I meant to investigate this, but you got to it before me. Thanks 😄

**Comment by franky47** [comment]
Hey @trevorblades, as a heads-up, I'm changing the way the Image component is included.

In `3.0.0-beta.1`, it was included at the top-level with the rest, but this has a major drawback: it automatically opts-in to the Next.js Image Optimization feature, even if no Image component is actually used (simply importing '@47ng/chakra-next' made static exports impossible for example, as Image Optimization requires a server or an image loader).

So I'm moving it out of the top-level, the component can be accessed in `3.0.0-beta.2` the following way:

```ts
import { Image } from '@47ng/chakra-next/dist/components/image'
```

**Pull request closed without merging**